src/directives/compose_area.ts

```diff
@@ -735,6 +735,9 @@ export default [
                     });
                     stopTyping();
                 });
+
+                // When losing browser or tab focus, send stopTyping message
+                window.onblur = () => stopTyping();
```

Comment:
Overwriting `onblur` is not an ideal solution because it might overwrite other event handlers that want to listen to the `blur` event. Furthermore, the event is never deregistered, so the `blur` event will trigger even if no chat is open.

The better solution would be to use the `unsubscribeListeners` hack used above:

- Store the callback in a local variable: `const stopTypingOnBlur = () => stopTyping()`
- To register the event listener, use `window.addEventListener('blur', stopTypingOnBlur)`
- To ensure that the listener is deregistered properly, push `() => window.removeEventListener('blur', stopTypingOnBlur)` into `unsubscribeListeners`

This is all untested, so please check whether it works as intended :slightly_smiling_face:

Reply:
Resolved in 50060caa21295ca4e7f79195d4dd2837a8973fd9.